setup_unit_tests.py

```diff
@@ -0,0 +1,488 @@
+                        except Exception as e:
+                            raise Exception(
+                                f"Error writing files for example {ex}"
+                            ) from e
+
+    all_data_files = None
+    if data_dir is not None:
+        all_data_files = set(os.path.basename(x) for x in glob.glob(str(data_dir / "**"), recursive=True))
+
+    if data_dir is not None:
+        output_data_dir = output_dir / "data"
+    else:
+        output_data_dir = None
+
+    metadata = None
+    if extra_metadata is not None:
+        with open(extra_metadata, "r") as e:
+            yaml = YAML()
+            metadata = yaml.load(e)
+
+    for m in all_m:
+        generate_config_file(m, output_dir, version, all_data_files, data_dir, output_data_dir, config, metadata)
+
+    if output_type == "json":
+        config_file = output_dir / "test_config.json"
+        with open(config_file, "w") as o:
+            json.dump(config, o, indent=2)
+    else:
+        config_file = output_dir / "test_config.yaml"
+        with open(config_file, "w") as o:
+            yaml = YAML()
+            yaml.dump(config, o)
+
+    if data_dir is not None and data_dir.exists():
+        shutil.copytree(data_dir, output_data_dir, symlinks=True, dirs_exist_ok=False)
+
+
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv[1:]
+    parser = argparse.ArgumentParser(
+        usage="%(prog)s [options]",
+        description=(
+            "Extracts the unit tests from the spec to be runnable by this test suite;"
+            "test files will be outputted into the `unit_tests` directory and a new conformance file will be created."
+            "To run the unit tests, see [insert other script here]"
+        )
+    )
+    parser.add_argument(
+        "--version",
+        "-v",
+        default="1.1",
+        help="Version of the spec to grab unit tests out of. Only WDL 1.1 and above has support for built in unit tests.",
+        choices=["1.1", "1.2"]
+    )
+    parser.add_argument(
+        "--output-type",
+        default="yaml",
+        choices=["yaml", "json"],
+        help="Specify conformance file output type."
+    )
+    parser.add_argument(
+        "--force-pull",
+        default=False,
+        action="store_true",
+        help="Default behavior does not pull the spec repo if it was already pulled. Specify this argument to force a refresh."
+    )
+    parser.add_argument(
+        "--extra-metadata", "-e",
+        default="unit_tests_metadata.yaml",
+        help="Include extra metadata when extracting the unit tests into the config. "
+             "Since we have our own method of testing file outputs, this is mostly used for file hashes/regexes."
+    )
+    parser.add_argument(
+        "--script",
+        # default="unit_tests_script.sh",
+        default=None,
+        help="Bash script to run alongside. This is to set up the environment for the test suite, ex mount points and files tests may depend on."
```

Comment:
What does "alongside" mean? Where and when does this run?

Reply:
I changed the wording and moved it to `run_unit.py`.